modAionImpl/src/org/aion/zero/impl/sync/TaskFastImportBlocks.java

```diff
@@ -0,0 +1,122 @@
+package org.aion.zero.impl.sync;
+
+import java.util.List;
+import org.aion.mcf.core.ImportResult;
+import org.aion.types.ByteArrayWrapper;
+import org.aion.zero.impl.AionBlockchainImpl;
+import org.aion.zero.impl.types.AionBlock;
+import org.slf4j.Logger;
+
+/**
+ * Task for importing blocks into the repository when running fast sync. The blocks are processed
+ * different from {@link TaskImportBlocks} because fewer consensus validations are applied when
+ * running fast sync. Also, the blocks are imported in reverse order starting from the oldest
+ * ancestor back to the first block.
+ *
+ * @author Alexandra Roatis
+ */
+final class TaskFastImportBlocks implements Runnable {
+
+    private final AionBlockchainImpl chain;
+    private final FastSyncManager fastSyncMgr;
+    private ByteArrayWrapper required;
+    private final Logger log;
+
+    TaskFastImportBlocks(
+            final AionBlockchainImpl chain, final FastSyncManager fastSyncMgr, final Logger log) {
+        this.chain = chain;
+        this.fastSyncMgr = fastSyncMgr;
+        this.log = log;
+        this.required = null;
+    }
+
+    @Override
+    public void run() {
+        Thread.currentThread().setPriority(Thread.MAX_PRIORITY);
```

Comment:
Don't agree to use the MAX_PRIORITY here since the fastsync mgr use all available threads to execute the block import. It might affect the other p2p in/out threads then spend more time waiting for the data.

Reply:
The reason why this was set to max priority is that I was following the same structure as for `TaskImportBlocks`. The fast sync manager threads are actually used for receiving and validating blocks using `TaskValidateAndAddBlocks` which has a lower priority. However, until I add all the functionality for fast sync and can make a better decision, I will set all the thread to `Thread.NORM_PRIORITY`.